### PR TITLE
Add demo of AnimatedSwitch and AnimatedButton to Example app

### PR DESCRIPTION
**Pull request by calda**
This PR adds a demo of `AnimatedSwitch` and `AnimatedButton` to the Example app. These don't currently work correctly with the experimental rendering engine -- will work on fixes in a follow-up.

### CODE REVIEW

File: Example/iOS/ViewControllers/ControlsDemoViewController.swift [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    super.init(layout: layout)
+    setItems(items, animated: false)
+
+    title = "Controls Demo"
+  }
+
+  // MARK: Internal
+
+  var items: [ItemModeling] {
```
Comment (erichoracek) [inline Example/iOS/ViewControllers/ControlsDemoViewController.swift:25]: nit: if you'd like you can make this an `@ItemModelBuilder` to omit the commas and array literal

File: Example/iOS/Views/AnimatedButtonRow.swift [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+      else { return false }
+
+      for (lhsRanges, rhsRanges) in zip(lhs.playRanges, rhs.playRanges) {
+        if lhsRanges != rhsRanges {
+          return false
+        }
+      }
+
+      return true
+    }
```
Comment (erichoracek) [inline Example/iOS/Views/AnimatedButtonRow.swift:47]: nit: making `playRanges` a nested `Equatable` struct would probably be shorter than this implementation

File: Example/iOS/Views/AnimatedSwitchRow.swift [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    fatalError("init(coder:) has not been implemented")
+  }
+
+  // MARK: Internal
+
+  struct Content: Equatable {
+    var animationName: String
+    var title: String
+    var onTimeRange: (from: CGFloat, to: CGFloat)
+    var offTimeRange: (from: CGFloat, to: CGFloat)
```
Comment (erichoracek) [inline Example/iOS/Views/AnimatedSwitchRow.swift:33]: How about using `ClosedRange<CGFloat>` instead? Will that also allow us to remove our equatable implementation?

File: Example/iOS/ViewControllers/SampleListViewController.swift [hunk 4]
```diff
@@ -83,13 +83,28 @@ final class SampleListViewController: CollectionViewController {
...
+    if isTopLevel {
+      items += [
+        LinkView.itemModel(
+          dataID: "Controls Demo",
+          content: .init(animationName: nil, title: "Controls Demo"))
+          .didSelect { [weak self] context in
+            self?.show(ControlsDemoViewController(), sender: context.view)
+          },
+      ]
+    }
```
Comment (erichoracek) [inline Example/iOS/ViewControllers/SampleListViewController.swift:95]: nit: Suggest breaking this out into multiple methods that return arrays of items models for readability

**Review by erichoracek — APPROVED** [review]
LGTM! Just a few small nitpicks

File: Example/iOS/Views/AnimatedButtonRow.swift (hunk 2 above)
Comment (calda) [reply Example/iOS/Views/AnimatedButtonRow.swift:47]: good idea

File: Example/iOS/Views/AnimatedSwitchRow.swift (hunk 3 above)
Comment (calda) [reply Example/iOS/Views/AnimatedSwitchRow.swift:33]: Good idea!

File: Example/iOS/ViewControllers/SampleListViewController.swift (hunk 4 above)
Comment (calda) [reply Example/iOS/ViewControllers/SampleListViewController.swift:61]: Good suggestion, much better now https://github.com/airbnb/lottie-ios/pull/1462/commits/48f98df7802f3b9a5693275bb06088815cc00968

**Pull request merged**